infra/license/license-checker.ts

```diff
@@ -31,6 +31,129 @@ var args = process.argv.slice(2)
+
+  failForbiddenLicense: 'Forbidden license is found',
+  failForbiddenPackage: 'Forbidden package is used'
+};
+type VerificationResultTypeKeys = keyof typeof VerificationResultType;
+
+type VerificationResult = {
+  pkgName: string; pkgLicense: string; verificationResult: VerificationResultTypeKeys;
+};
+
+class VerificationResultSet {
+  private results: VerificationResult[];
+
+  constructor() {
+    this.results = [] as VerificationResult[];
+  };
+
+  public add(result: VerificationResult) {
+    this.results.push(result);
+  };
+
+  public warnCount(): number {
+    let count = 0;
+    this.results.forEach((result) => {
+      if (result.verificationResult.startsWith('warn')) {
+        count++;
+      }
+    });
+    return count;
+  };
+
+  public failCount(): number {
+    let count = 0;
+    this.results.forEach((result) => {
+      if (result.verificationResult.startsWith('fail')) {
+        count++;
+      }
+    });
+    return count;
+  };
+
+  public createComment(): string {
+    const title = '#### ' + verificationTag + EOL + EOL;
+
+    let body = '';
+    if (this.warnCount() + this.failCount() === 0) {
+      body += (':heavy_check_mark: No license issue found' + EOL);
+    }
+    if (this.warnCount() > 0) {
+      body += (':warning: **Warning** :warning:' + EOL);
+      for (const vtype in VerificationResultType) {
+        if (vtype.startsWith('warn')) {
+          body += this.createDetailedComment(vtype as VerificationResultTypeKeys);
+        }
+      }
+    }
+    if (this.failCount() > 0) {
+      body += (':no_entry: **Failure** :no_entry:' + EOL);
+      for (const vtype in VerificationResultType) {
+        if (vtype.startsWith('fail')) {
+          body += this.createDetailedComment(vtype as VerificationResultTypeKeys);
+        }
+      }
+    }
+
+    return title + body;
+  };
+
+  private createDetailedComment(vtype: VerificationResultTypeKeys): string {
+    let title = '';
+    let body = '';
+
+    this.results.forEach((result) => {
+      if (result.verificationResult === vtype) {
+        title = '- ' + VerificationResultType[vtype] + EOL;
+        body += ('    - ' + result.pkgName + ' : ' + result.pkgLicense + EOL);
+      }
+    });
+
+    return title + body;
+  };
+};
+
+function verify(pkgName: string, pkgLicense: string): VerificationResultTypeKeys {
+  if (packageJudgment.hasOwnProperty(pkgName)) {
+    switch (packageJudgment[pkgName as keyof typeof packageJudgment].permitted) {
+      case 'yes':
+        return 'pass';
+      case 'conditional':
+        return 'pass';
+      case 'no':
+        return 'failForbiddenPackage';
+      default:
+        throw new Error('Not implemented permitted type');
+    }
+  }
+
+  if (licenseJudgment.hasOwnProperty(pkgLicense)) {
+    switch (licenseJudgment[pkgLicense as keyof typeof licenseJudgment].permitted) {
+      case 'yes':
+        return 'pass';
+      case 'conditional':
+        return 'warnConditionalLicense';
```

Comment:
It would be better if the literal strings like 'yes', 'warnConditionalLicense' are extracted as a kind of enum.

Reply:
It's possible to cause errors from typo.

Reply:
https://github.com/Samsung/ONE-vscode/pull/393#discussion_r832777862 would resolve this question :D